fe/src/pages/user/register/Register.jsx

```diff
@@ -0,0 +1,109 @@
+import React, { useState } from 'react';
+import axios from 'axios';
+import Modal from '../../../components/Modal';
+
+const Register = () => {
+  const [formData, setFormData] = useState({
+    fullname: '',
+    email: '',
+    password: '',
+  });
+  const [errorPassword, setErrorPassword] = useState('');
+  const [errorEmail, setErrorEmail] = useState('');
+  const [showModal, setShowModal] = useState(false);
+  const handleClose = () => {
+    setShowModal(false);
+  };
+
+  const handleChange = (event) => {
+    setFormData({
+      ...formData,
+      [event.target.name]: event.target.value,
+    });
+  };
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    if (formData.password !== formData.confirmPassword) {
+      setErrorPassword('Passwords do not match');
+    } else {
+      setErrorPassword('');
+      axios.post('http://127.0.0.1:8000/api/register', {
+        ...formData,
+      })
+        .then(() => {
+          window.location.pathname = '/';
+        })
+        .catch((error) => {
+          if (error.response.status === 422) {
+            setErrorEmail('The Email has Already Taken');
+          }
+        });
+    }
+  };
```

Comment:
- separate this to a hook to avoid overcrowding in the component. (refer https://www.youtube.com/watch?v=Jl4q2cccwf0)